aeron-archive/src/test/java/io/aeron/archive/ArchiveMigrationUtils.java

```diff
@@ -0,0 +1,319 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.aeron.archive;
+
+import io.aeron.archive.client.ArchiveException;
+import org.agrona.IoUtil;
+import org.agrona.LangUtil;
+import org.agrona.SemanticVersion;
+import org.agrona.concurrent.EpochClock;
+import org.agrona.concurrent.UnsafeBuffer;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static io.aeron.archive.Catalog.MIN_CAPACITY;
+import static java.nio.ByteOrder.LITTLE_ENDIAN;
+import static java.nio.channels.FileChannel.MapMode.READ_WRITE;
+import static java.nio.file.StandardOpenOption.*;
+import static org.agrona.BitUtil.SIZE_OF_INT;
+import static org.agrona.BitUtil.SIZE_OF_LONG;
+
+final class ArchiveMigrationUtils
+{
+    static final int VERSION_2_1_0 = SemanticVersion.compose(2, 1, 0);
+
+    private static final int MARK_FILE_HEADER_LENGTH_V2 = 8 * 1024;
+    static final int RECORDING_FRAME_LENGTH_V2 = 1024;
+
+    private ArchiveMigrationUtils()
+    {
+    }
+
+    static ArchiveMarkFile createArchiveMarkFileInVersion2(final File archiveDir, final EpochClock epochClock)
+    {
+        final Path markFile = archiveDir.toPath().resolve(ArchiveMarkFile.FILENAME);
+        try (FileChannel channel = FileChannel.open(markFile, CREATE_NEW, READ, WRITE, SPARSE))
+        {
+            final int errorBufferLength = 100;
+            final String controlChannel = "ctx.controlChannel()";
+            final String localControlChannel = "ctx.localControlChannel()";
+            final String eventsChannel = "ctx.recordingEventsChannel()";
+            final String aeronDirectory = "ctx.aeronDirectoryName()";
+
+            final MappedByteBuffer mappedByteBuffer = channel.map(
+                READ_WRITE,
+                0,
+                totalMarkFileLengthInVersion2(
+                controlChannel, localControlChannel, eventsChannel, aeronDirectory, errorBufferLength));
+            mappedByteBuffer.order(LITTLE_ENDIAN);
+            try
+            {
+                final UnsafeBuffer buffer = new UnsafeBuffer(mappedByteBuffer);
+                final long currentTime = epochClock.time();
+
+                int index = 0;
+                buffer.putInt(index, VERSION_2_1_0); // version
+                index += SIZE_OF_INT;
+
+                buffer.putLong(index, currentTime, LITTLE_ENDIAN); // activityTimestamp
+                index += SIZE_OF_LONG;
+
+                buffer.putLong(index, currentTime, LITTLE_ENDIAN); // startTimestamp
+                index += SIZE_OF_LONG;
+
+                buffer.putLong(index, 1, LITTLE_ENDIAN); // pid
```

Comment:
The use of magic numbers here is not good.

Reply:
Agree. Those can be pulled out either as constants or better yet as parameters to this method. This class is used only in tests.

Reply:
The comments are a smell.

Reply:
I'll refactor it.